submissions/eoneoff/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env node
+
+"use strict";
+
+let help = {
+    script:`Simple port sniffer
+
+This is a simple port sniffer, which can be calles as command line script or imported as a module.
+When used from command line it takes 2 named paramentes:
+--host      obligatory parameter which defines the host address for a port scan.
+            After the --host parameter should follow the host IP addres as four numbers,
+            separated by commas or a host web address.
+
+                python sniffer.py --host www.google.com
+
+--ports     optional parementer to provide a port range for scan. The range
+            should be definded as two number with a dash between them. If not set
+            or set incorrectly the default range of (1-65535) is scanned
+
+                python sniffer.py --host wwww.google.com --ports 1024-2018
+
+--help      parameted to see this reference
+
+To import port scanner use
+
+    import sniffer.py
+
+after import you can perform scan by calling sniffer.scan()
+
+scan(host, portString, module, silent)
+
+For  detailed disambiguation of scan function use help(scan) after importing
+sniffer or call sniffer.py with --help method`
+}
+
+const dns = require('dns');
+const Socket = require('net').Socket;
+
+function* twister(){
+    let symbols = ['|','/','-','\\'];
+    let current = 0;
+    while (true) {
+        if (current == 4) current = 0;
+        yield symbols[current++];
+    }
+}
+
+async function parseHost(hostName){
+    let host = ''
+    if(hostName.search(/^((25[0-5]|2[0-4]\d|[01]?\d?\d)\.){3}(25[0-5]|2[0-4]\d|[01]?\d?\d)$/) == -1){
+        host = await new Promise((resolve, reject)=>{
+            dns.lookup(hostName, (error, address) => {
+                if(error){
+                    error.message = "Hostname can not be resolved";
+                    reject(error);
+                }
+                resolve(address)
+            });
+        });
+    } else {
+        host = hostName;
+    }
+    return host;
+}
+
+function parsePorts(ports){
+    if(ports.search(/\d{1,4}-\d{1,4}/) == -1){

```

Comment:
Same comment as for ipv4 check - better to abstract this logic to a helper function